packages/ui/src/components/button-with-options.tsx

```diff
@@ -99,12 +82,7 @@ export const ButtonWithOptions = <T extends string>({
       </Button>
       <DropdownMenu.Root>
         <DropdownMenu.Trigger
-          className={cn(
-            buttonVariants({ theme, variant }),
-            'relative h-[inherit] w-8 p-0 flex-shrink-0 rounded-l-none before:absolute before:left-0 before:h-[calc(100%-8px)] before:w-px',
-            theme !== 'primary' && 'border-y border-r',
-            separatorThemes[disabled ? 'disabled' : theme || 'default']
-          )}
+          className={cn(buttonVariants({ theme, variant }), 'button-split-dropdown')}
```

Comment:
same as above, can put an `asChild` and render the Button component here instead of adding button classes on a trigger?

Reply:
same here. Internally it is using button component. 
<img width="846" alt="image" src="https://github.com/user-attachments/assets/e220d80a-8094-446e-8adb-a7fd0be18cd3" />